addons/membership/models/partner.py

```diff
@@ -126,7 +126,7 @@ def _membership_state(self):
                                 s = 3
                 if s == 4:
                     for mline in partner.member_lines:
-                        if mline.date_from < today and mline.date_to < today and mline.date_from <= mline.date_to and mline.account_invoice_line and mline.account_invoice_line.invoice_id.state == 'paid':
+                        if (mline.date_from or '0000-00-00') < today and (mline.date_to or '0000-00-00') < today and (mline.date_from or '0000-00-00') <= (mline.date_to or '0000-00-00') and mline.account_invoice_line and mline.account_invoice_line.invoice_id.state == 'paid':
```

Comment:
Maybe create aliases for these values to avoid having to do the same thing multiple times in a row?